state/benches/applicator.rs

```diff
@@ -0,0 +1,132 @@
+//! Benchmarks the state applicator
+#![allow(missing_docs, clippy::missing_docs_in_private_items)]
+use std::time::{Duration, Instant};
+
+use common::types::{
+    gossip::WrappedPeerId,
+    tasks::{mocks::mock_queued_task, QueuedTaskState},
+    wallet_mocks::mock_empty_wallet,
+};
+use criterion::{criterion_group, criterion_main, Criterion, Throughput};
+use state::{
+    applicator::{test_helpers::mock_applicator, StateApplicator},
+    StateTransition,
+};
+use uuid::Uuid;
+
+/// Create a mock applicator with necessary DB entries filled in
+fn benchmark_applicator() -> StateApplicator {
+    let applicator = mock_applicator();
+    let tx = applicator.db().new_write_tx().unwrap();
+    let peer_id = WrappedPeerId::random();
+    tx.set_peer_id(&peer_id).unwrap();
+    tx.commit().unwrap();
+
+    applicator
+}
+
+/// Benchmark updating a wallet
+fn bench_update_wallet(c: &mut Criterion) {
+    let applicator = benchmark_applicator();
+    let wallet = mock_empty_wallet();
+    let transition = StateTransition::UpdateWallet { wallet };
+
+    let mut group = c.benchmark_group("applicator");
+    group.throughput(Throughput::Elements(1));
+    group.bench_function("update_wallet", |b| {
+        b.iter_custom(|n_iters| {
+            let mut total_time = Duration::default();
+            for _ in 0..n_iters {
+                let start = Instant::now();
+                applicator.handle_state_transition(Box::new(transition.clone())).unwrap();
+                total_time += start.elapsed();
+            }
+            total_time
+        });
+    });
+}
```

Comment:
how useful is this benchmark to us if it tests transitioning an empty wallet to another empty wallet? there's lots of interesting conditional logic e.g. indexing new & cancelled orders which will never be invoked here but is common in practice

Reply:
yeah agreed we should test those things, though at the moment these benchmarks tell us that serialization of large arrays is the bottleneck so currently it's sufficient for highlighting the bottleneck apart from any conditional logic